integreat_cms/api/v3/socialmedia_headers.py

```diff
@@ -0,0 +1,298 @@
+"""
+This module includes functions related to the social media headers API endpoint.
+"""
+from __future__ import annotations
+
+import logging
+from html import unescape
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.shortcuts import render
+from django.utils.html import strip_tags
+from django.utils.translation import activate
+
+from integreat_cms.cms.models.events.event_translation import EventTranslation
+from integreat_cms.cms.models.languages.language import Language
+from integreat_cms.cms.models.pois.poi_translation import POITranslation
+from integreat_cms.cms.models.push_notifications.push_notification_translation import (
+    PushNotificationTranslation,
+)
+from integreat_cms.cms.models.regions.region import Region
+
+from ...cms.models import PageTranslation
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from django.http import HttpRequest, HttpResponse
+
+logger = logging.getLogger(__name__)
+
+
+def socialmedia_headers(
+    request: HttpRequest, path: Optional[str] = None
+) -> HttpResponse:
+    """
+    Detects the path context and renders the socialmedia headers.
+    Available contexts are:
+    - root e.g. /, /landing, /disclaimer, /landing/de
+    - region root e.g. /testumgebung, /testumgebung/de, /testumgebung/de/events
+    - page e.g. /testumgebung/de/willkommen, /testumgebung/de/events/event
+
+    :param request: The request that has been sent to the Django server
+    :param path: The path of the page the social meddia headers should be created for
+
+    :return: HTML meta headers required by social media platforms
+    """
+    slugs = list(filter(bool, path.split("/"))) if path else []
+
+    potential_region_slug = slugs[0] if len(slugs) > 0 else None
+    region = (
+        Region.objects.filter(slug=potential_region_slug).first()
+        if potential_region_slug
+        else None
+    )
+
+    potential_language = (
+        Language.objects.filter(slug=slugs[1]).first() if len(slugs) > 1 else None
+    )
+    language = potential_language or Language.objects.get(slug="de")
+    if not region:
+        return root_socialmedia_headers(request, language, path or "")
```

Comment:
maybe we could check for reserved urls and then return 404. Otherwise https://webnext.integreat.app/recommend/de, which is imo a very good page might not get a good (search result) score.

Reply:
Yes, I agree. Let's show the generic fallback preview for the special reserved slugs, and only handle real 404 errors differently.